index.bs

```diff
@@ -2,16 +2,715 @@
+
+Commands {#protocol-commands}
+-----------------------------
+
+A <dfn>command</dfn> is an asynchronous operation, requested by the [=local end=] and run on the [=remote end=], resulting in either a [=success=] or an [=error=] being returned to the [=local end=]. Multiple commands can run at the same time, and commands can potentially be long-running. As a consequence, commands can finish out-of-order.
+
+Each [=command=] is defined by:
+
+* A <dfn>command type</dfn> which is defined by a [=remote end definition=] fragment containing a group. Each such group has two fields:
+    * `method` which is a string literal in the form `[module name].[method name]`. This is the <dfn>command name</dfn>.
+    * `params` which defines a mapping containing data that to be passed into the command. The populated value of this map is the <dfn>command parameters</dfn>.
+* A <dfn>result type</dfn>, which is defined by the [=local end definition=] fragment.
+* A set of <dfn>remote end steps</dfn> which define the actions to take for a command given a [=session=] and [=command parameters=] and return an instance of the command [=return type=].
+
+A command that can run without an active session is a <dfn>static command</dfn>. Commands are not static commands unless stated in their definition.
+
+When commands are sent from the [=local end=] they have a command id. This is an identifier used by the [=local end=] to identify the response from a particular command. From the point of view of the [=remote end=] this identifier is opaque and cannot be used internally to identify the command.
+
+The <dfn>set of all command names</dfn> is a [=set=] containing all the defined [=command names=], including any belonging to [=extension modules=].
+
+Events {#protocol-events}
+-------------------------
+
+An <dfn>event</dfn> is a notification, sent by the [=remote end=] to the [=local end=], signaling that something of interest has occurred on the [=remote end=].
+
+* An <dfn>event type</dfn> is defined by a [=local end definition=] fragment containing a group. Each such group has two fields:
+    * `method` which is a string literal of the form `[module name].[event name]`. This is the <dfn>event name</dfn>.
+    * `params` which defines a mapping containing event data. The populated value of this map is the <dfn>event parameters</dfn>.
+* A <dfn>remote end event trigger</dfn> which defines when the event is triggered and steps to construct the [=event type=] data.
+
+Errors {#protocol-errors}
+-------------------------
+
+The following table lists each <dfn>error code</dfn>, its associated JSON `error` code, and a non-normative description of the error.
+
+<table>
```

Comment:
Given each `Error code` is identical to its corresponding `JSON error code`, is this distinction necessary? Can we consider hypotheticals where this is not the case?

Reply:
I did it this way to match [WebDriver](https://w3c.github.io/webdriver/#dfn-error-code), except we don't have HTTP response codes.

The table makes the casing explicit, at least, so I think we should keep it even if the names and the strings always match.